### DISCUSSION TITLE
edk build windows

### DISCUSSION THREAD

**Discussion by joe-cryptosecurellc** [tianocore/edk2#3831] [Q&A]
I cannot build with the edk on windows with openssl. My library will not build because it cannot find the openssl headers.
My library is part of the Security Package. My inf file for my library includes CryptoPkg.dec in the [Packages] list and OpensslLib in the [LibraryClasses] list. The CryptoPkg.dec includes the OpensslLib path in the [Includes.Common.Private] list as "Library/OpensslLib/openssl/crypto/include". My edk is a commit branch from Dec 10, 2019, I gor openssl using "git submodule update --init.

My library source files reference openssl headers as "#include <openssl.aes.h>"

**Comment by joe-cryptosecurellc** [comment]
Sorry - I forgot to mention that the same build works on Ubuntu 20.04.

**Comment by trhpx** [comment]
For any include path with .Private like this, the build tools create the concept of a library path that is only available for modules within the same package, in this case, CryptoPkg.



[Relevant DEC file explanation](https://edk2-docs.gitbook.io/edk-ii-dec-specification/2_dec_file_overview/25_-includes-_usage)

> The section tag modifier, Private, is used to restrict the EDK II build system by preventing references to content in these sections from being used by modules outside of the package.
